universal-login-sdk/lib/services/SecurityCodeService.ts

```diff
@@ -0,0 +1,63 @@
+import { utils } from 'ethers';
+import deepEqual from 'deep-equal';
+import { InvalidAddress } from '../utils/errors';
+
+export class SecurityCodeService {
+  maxCodeNumber: number;
+  securityCodeLength: number;
+  securityKeyboardSize: number;
+  mask: number;
+  highNumberMask: number;
+  lowNumberMask: number;
+
+  constructor() {
+    this.maxCodeNumber = 1024;
+    this.securityCodeLength = 6;
+    this.securityKeyboardSize = 30;
+    this.mask = 0x03FF;
+    this.highNumberMask = 0x03;
+    this.lowNumberMask = 0xFF;
+  }
+
+  encode(address: string): number[] {
+    if (!address.match(/0x[0-9A-Fa-f]{40}/)) {
+      throw new InvalidAddress(address);
+    }
+    const addressBytes = utils.arrayify(address);
+    const code = [];
+    for (let i = 0; i !== addressBytes.length; i += 2) {
+      code.push(this.to10bitNumber(addressBytes[i], addressBytes[i + 1]));
+    }
+
+    return code
+      .slice(0, this.securityCodeLength)
+      .map((codeElement) => codeElement & 0x03FF);
```

Comment:
`.map((codeElement) => codeElement & 0x03FF)` 
-> 
`.map((codeElement) => codeElement & this.mask)`